src/Chainweb/TreeDB.hs

```diff
@@ -716,6 +721,51 @@ branchDiff_ db = go
             go lp rp
     step = lift . lookupParentM GenesisParentThrow db
 
+-- -------------------------------------------------------------------------- --
+-- | Walks the ancestry of the given tree from the leaf back to the genesis.
+streamAncestors
+    :: forall db . TreeDb db
+    => db
+    -> DbEntry db
+    -> S.Stream (Of (DbEntry db)) IO ()
+streamAncestors db = go
```

Comment:
isn't this a special case of `branchEntries`? 

```Haskell
f (streamAncestors db e) == branchEntries db Nothing Nothing Nothing  Nothing mempty (HS.singleton $ UpperBound $ key e) f
```

The advantage of the implementation of `streamAncestors` in the PR is that it returns a stream that is safe to use outside of a bracket.

The disadvantage is that it prevents more efficient implementations for instances that require the bracket for resource management (like, for instance, implementations via rocksdb iterators, which is on our TODO list).

If the motivation is to provide a stream without a CPS bracket, maybe explain it in the haddock comment.

Reply:
I ended up not needing this function so I removed it